### ISSUE TITLE
不知道为什么网络好像没在训练。

### ISSUE THREAD

**Issue by Lkinyuu** [Oneflow-Inc/oneflow#3967]
训练了22个epoch的输出，准确率和loss的变化看起来完全没在训练。 学习率是0.0005应该也不是学习率的问题
Epoch1 Time:243.7s Data:130.5s Loss:21.1768 Xent:6.8184 Htri:14.3583 Acc:0.80% 
Epoch2 Time:237.9s Data:120.0s Loss:19.0994 Xent:6.6358 Htri:12.4636 Acc:0.82% 
Epoch3 Time:237.9s Data:120.4s Loss:18.9156 Xent:6.6559 Htri:12.2598 Acc:0.69% 
Epoch4 Time:238.0s Data:120.0s Loss:18.9751 Xent:6.6308 Htri:12.3442 Acc:0.74% 
Epoch5 Time:238.1s Data:120.4s Loss:18.9926 Xent:6.6339 Htri:12.3587 Acc:0.76% 
Epoch6 Time:238.0s Data:120.0s Loss:19.0231 Xent:6.6509 Htri:12.3721 Acc:0.85% 
Epoch7 Time:238.1s Data:121.5s Loss:19.0045 Xent:6.6361 Htri:12.3684 Acc:0.97% 
Epoch8 Time:238.0s Data:123.0s Loss:19.0336 Xent:6.6219 Htri:12.4117 Acc:0.81% 
Epoch9 Time:238.0s Data:121.2s Loss:19.0933 Xent:6.6224 Htri:12.4709 Acc:0.89% 
Epoch10 Time:238.0s Data:121.4s Loss:18.7353 Xent:6.6354 Htri:12.0998 Acc:0.70% 
Epoch11 Time:238.0s Data:122.8s Loss:18.8023 Xent:6.6092 Htri:12.1931 Acc:1.00% 
Epoch12 Time:238.0s Data:122.3s Loss:18.8836 Xent:6.6332 Htri:12.2505 Acc:0.74% 
Epoch13 Time:238.4s Data:124.5s Loss:18.8834 Xent:6.6321 Htri:12.2512 Acc:0.82% 
Epoch14 Time:238.1s Data:124.8s Loss:18.7013 Xent:6.6213 Htri:12.0799 Acc:0.93% 
Epoch15 Time:238.0s Data:125.4s Loss:18.6192 Xent:6.6387 Htri:11.9806 Acc:0.73% 
Epoch16 Time:238.1s Data:126.4s Loss:18.6850 Xent:6.6161 Htri:12.0690 Acc:0.85% 
Epoch17 Time:238.1s Data:126.0s Loss:18.7358 Xent:6.6138 Htri:12.1220 Acc:0.84% 
Epoch18 Time:238.1s Data:125.8s Loss:18.8409 Xent:6.6212 Htri:12.2197 Acc:0.84% 
Epoch19 Time:238.1s Data:125.9s Loss:18.4976 Xent:6.6139 Htri:11.8837 Acc:1.01% 
Epoch20 Time:238.1s Data:125.4s Loss:18.4484 Xent:6.6040 Htri:11.8444 Acc:0.78% 
Epoch21 Time:238.1s Data:125.1s Loss:18.5733 Xent:6.6036 Htri:11.9697 Acc:0.70% 
Epoch22 Time:238.2s Data:125.6s Loss:18.7198 Xent:6.6030 Htri:12.1168 Acc:0.90%

**Comment by daquexian** [comment]
请问你在训练的是什么网络，使用的是什么优化器，在什么数据集上呢

**Comment by Lkinyuu** [comment]
> 请问你在训练的是什么网络，使用的是什么优化器，在什么数据集上呢
训练的是以Resnet50为骨干网络的一个用于ReID的网络。
使用的优化器：
r_scheduler=flow.optimizer.PiecewiseScalingScheduler(
            base_lr=args.lr,boundaries=args.stepsize,
            scale=[0.1,0.01,0.001]
        )
    flow.optimizer.AdamW(lr_scheduler, do_bias_correction=False,weight_decay=args.weight_decay).minimize(loss)

数据集：MARS

**Comment by MARD1NO** [comment]
![image](https://user-images.githubusercontent.com/42901638/100994652-c5344800-3591-11eb-8da0-a10ec6e2db31.png)
为啥准确率会有1.01%，是不是指标算错了？看Loss是有逐步在下降的，可以多跑几轮，也可以换成其他优化器如Adam试试

**Comment by Lkinyuu** [comment]
> ![image](https://user-images.githubusercontent.com/42901638/100994652-c5344800-3591-11eb-8da0-a10ec6e2db31.png)
> 为啥准确率会有1.01%，是不是指标算错了？看Loss是有逐步在下降的，可以多跑几轮，也可以换成其他优化器如Adam试试

2个loss外加准确率 也是一个在上下浮动的状态

**Comment by Lkinyuu** [comment]
> ![image](https://user-images.githubusercontent.com/42901638/100994652-c5344800-3591-11eb-8da0-a10ec6e2db31.png)
> 为啥准确率会有1.01%，是不是指标算错了？看Loss是有逐步在下降的，可以多跑几轮，也可以换成其他优化器如Adam试试

是1.01%又不是1.01，模型相当于完全没有在拟合，只是在随机输出

**Comment by doombeaker** [comment]
请问你是把 MARS 先转成了numpy再喂给oneflow里的模型的吧？你有没有先验证转成numpy时得到的数据是否是正确的呢？
有一种可能是，解析数据错了。

**Comment by Lkinyuu** [comment]
> 请问你是把 MARS 先转成了numpy再喂给oneflow里的模型的吧？你有没有先验证转成numpy时得到的数据是否是正确的呢？
> 有一种可能是，解析数据错了。

这个可能似乎已经被排除了，因为利用numpy的时候已经注意了HWC和CHW的问题。而且训练效果已经差到 直接对数据reshape都没什么大变化了。

**Comment by doombeaker** [comment]
在脚本开始时加入

**Comment by xunhen** [comment]
你好，问一下，你的问题解决了吗
